Content.Client/Storage/Visualizers/StorageVisualizerSystem.cs

```diff
@@ -0,0 +1,93 @@
+using Content.Shared.Storage;
+using Robust.Client.GameObjects;
+
+namespace Content.Client.Storage.Visualizers;
+
+public sealed class StorageVisualizerSystem : VisualizerSystem<StorageVisualizerComponent>
+{
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<StorageVisualizerComponent, ComponentInit>(OnInit);
+    }
+
+    /// <summary>
+    /// Sets the base sprite to this layer. Exists to make the inheritance tree less boilerplate-y.
+    /// </summary>
+    private void OnInit(EntityUid uid, StorageVisualizerComponent comp, ComponentInit args)
+    {
+        if(!TryComp<SpriteComponent>(uid, out var sprite))
+            return;
+
+        if (comp.StateBase != null)
+        {
+            sprite.LayerSetState(0, comp.StateBase);
+        }
+
+        if (comp.StateBaseAlt == null)
+        {
+            comp.StateBaseAlt = comp.StateBase;
+        }
+    }
+
+    protected override void OnAppearanceChange(EntityUid uid, StorageVisualizerComponent comp, ref AppearanceChangeEvent args)
+    {
+        if (args.Sprite == null)
+            return;
+        if(!AppearanceSystem.TryGetData(uid, StorageVisuals.Open, out bool open, args.Component))
+            return;
+
+        if (args.Sprite.LayerMapTryGet(StorageVisualLayers.Door, out _))
+        {
+            args.Sprite.LayerSetVisible(StorageVisualLayers.Door, true);
+
+            if (open)
+            {
+                if (comp.StateOpen != null)
+                {
+                    args.Sprite.LayerSetState(StorageVisualLayers.Door, comp.StateOpen);
+                    args.Sprite.LayerSetVisible(StorageVisualLayers.Door, true);
+                }
+
+                if (comp.StateBaseAlt != null)
+                    args.Sprite.LayerSetState(0, comp.StateBaseAlt);
+            }
+            else if (!open)
+            {
+                if (comp.StateClosed != null)
+                    args.Sprite.LayerSetState(StorageVisualLayers.Door, comp.StateClosed);
+                else
+                {
+                    args.Sprite.LayerSetVisible(StorageVisualLayers.Door, false);
+                }
+
+                if (comp.StateBase != null)
+                    args.Sprite.LayerSetState(0, comp.StateBase);
+            }
+            else
+            {
+                args.Sprite.LayerSetVisible(StorageVisualLayers.Door, false);
+            }
+        }
```

Comment:
i wont require you since its a bit out of scope but take a shot at making this part less.... verbose

Reply:
I couldn't do much, it will have to wait for a while.